dali/test/python/test_operator_slice.py

```diff
@@ -0,0 +1,225 @@
+import os
+
+from test_utils import check_batch
+from test_utils import compare_pipelines
+
+test_data_root = os.environ['DALI_EXTRA_PATH']
+caffe_db_folder = os.path.join(test_data_root, 'db', 'lmdb')
+test_data_video = os.path.join(test_data_root, 'db', 'optical_flow', 'sintel_trailer')
+
+class SlicePipeline(Pipeline):
+    def __init__(self, device, batch_size, pos_size_iter, num_threads=1, device_id=0, is_old_slice=True):
+        super(SlicePipeline, self).__init__(batch_size,
+                                            num_threads,
+                                            device_id,
+                                            seed=1234)
+        self.pos_size_iter = pos_size_iter
+        self.device = device
+        self.input = ops.CaffeReader(path = caffe_db_folder, random_shuffle=False)
+        self.input_crop_pos = ops.ExternalSource()
+        self.input_crop_size = ops.ExternalSource()
+        self.input_crop = ops.ExternalSource()
+        self.decode = ops.HostDecoder(device = "cpu", output_type = types.RGB)
+
+        if is_old_slice:
+            self.slice = ops.Slice(device = device,
+                                   image_type = types.RGB)
+        else:
+            self.slice = ops.NewSlice(device = device,
+                                      image_type = types.RGB)
+
+    def define_graph(self):
+        inputs, labels = self.input(name="Reader")
+        images = self.decode(inputs)
+        self.crop_pos = self.input_crop_pos()
+        self.crop_size = self.input_crop_size()
+        if self.device == 'gpu':
+            images = images.gpu()
+        out = self.slice(images, self.crop_pos, self.crop_size)
+        return out
+
+    def iter_setup(self):
+        (crop_pos, crop_size) = self.pos_size_iter.next()
+        self.feed_input(self.crop_pos, crop_pos)
+        self.feed_input(self.crop_size, crop_size)
+
+
+class SliceArgsIterator(object):
+    def __init__(self, batch_size):
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        self.i = 0
+        self.n = self.batch_size
+        return self
+
+    def __next__(self):
+        pos = []
+        size = []
+        for k in range(self.batch_size):
+            pos.append(np.asarray([0.4, 0.2], dtype=np.float32)) # xy
+            size.append(np.asarray([0.3, 0.5], dtype=np.float32)) # WH
+            self.i = (self.i + 1) % self.n
+        return (pos, size)
+    next = __next__
+
+def test_old_slice_vs_new_slice_gpu():
+    for batch_size in {1, 13, 64}:
+        eii1 = SliceArgsIterator(batch_size)
+        pos_size_iter1 = iter(eii1)
+
+        eii2 = SliceArgsIterator(batch_size)
+        pos_size_iter2 = iter(eii2)
+
+        compare_pipelines(SlicePipeline('gpu', batch_size, pos_size_iter1, is_old_slice=True),
+                          SlicePipeline('gpu', batch_size, pos_size_iter2, is_old_slice=False),
+                          batch_size=batch_size, N_iterations=10)
```

Comment:
Again, test generator would be useful this.

Reply:
addressed in #930